### PR TITLE
Change all API methods to POST request

### PR DESCRIPTION
**Pull request by sushantpatil1214**

### CODE REVIEW

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/ProxyRoutes.java [hunk 1]
```diff
@@ -458,43 +458,59 @@ static Route proxyPatchCrUpdateFields(
...
-   }
 
-   static Route proxyGetDashboardData(
+   static Route proxyPostCandidatesWithScore(
+        final String linkerIP,
+        final Integer linkerPort,
+        final Http http) {
+    return entity(Jackson.unmarshaller(ApiModels.ApiInteraction.class), request -> {
+        try {
+            return onComplete(proxyPostCandidatesWithScoreDoIt(linkerIP, linkerPort, http, request.uid()),
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/ProxyRoutes.java:492]: @sushantpatil1214 pass the entire request object not only the uid

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/ProxyRoutes.java [hunk 2]
```diff
@@ -458,43 +458,59 @@ static Route proxyPatchCrUpdateFields(
...
+        final String iid) {
+    // Construct the URI without query parameters
+         final var uri = String.format(Locale.ROOT,
+                  "http://%s:%d/JeMPI/%s",
+                  linkerIP,
+                  linkerPort,
+                  GlobalConstants.SEGMENT_PROXY_POST_CANDIDATES_WITH_SCORES);
+
+         // Construct the request body with the iid parameter
+         final var json = "{\"uid\": \"" + iid + "\"}";
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/ProxyRoutes.java:474]: @sushantpatil1214 use this implementation eg json = OBJECT_MAPPER.writeValueAsBytes(body);

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/ProxyRoutes.java [hunk 3]
```diff
@@ -458,43 +458,59 @@ static Route proxyPatchCrUpdateFields(
...
-                                    GlobalConstants.SEGMENT_PROXY_GET_CANDIDATES_WITH_SCORES,
-                                    iid);
-      final var request = HttpRequest.create(uri);
-      final var stage = http.singleRequest(request);
-      return stage.thenApply(response -> response);
+   private static CompletionStage<HttpResponse> proxyPostCandidatesWithScoreDoIt(
+        final String linkerIP,
+        final Integer linkerPort,
+        final Http http,
+        final String iid) {
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/ProxyRoutes.java:465]: @sushantpatil1214 update this to a request body class

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/ProxyRoutes.java [hunk 4]
```diff
@@ -458,43 +458,59 @@ static Route proxyPatchCrUpdateFields(
...
+                            if (e instanceof IllegalUriException illegalUriException) {
+                                LOGGER.error(illegalUriException.getLocalizedMessage(), illegalUriException);
+                            } else {
+                                LOGGER.error(e.getLocalizedMessage(), e);
+                            }
+                            return mapError(new MpiServiceError.InternalError(e.getLocalizedMessage()));
+                        }
+                        return complete(response.get());
+                    });
+        } catch (NumberFormatException e) {
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/ProxyRoutes.java:505]: @sushantpatil1214 where is the exception being thrown?

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java [hunk 5]
```diff
@@ -98,13 +96,13 @@ private static Route postIidGidLink(
    }
 
 
-   private static Route patchGoldenRecord(
+   private static Route postGoldenRecord(
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java:99]:
```suggestion
   private static Route updateGoldenRecord(
```

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java [hunk 6]
```diff
@@ -137,53 +135,65 @@ private static Route countRecords(
...
+            return onComplete(Ask.getGidsPaged(actorSystem, backEnd, request.offset(), request.length()),
+                    result -> {
+                        if (!result.isSuccess()) {
+                            final var e = result.failed().get();
+                            LOGGER.error(e.getLocalizedMessage(), e);
+                            return mapError(new MpiServiceError.InternalError(e.getLocalizedMessage()));
+                        }
+                        return complete(StatusCodes.OK, result.get(), JSON_MARSHALLER);
+                    });
+        } catch (NumberFormatException e) {
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java:152]: @sushantpatil1214 where is this exception type thrown?

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java [hunk 7]
```diff
@@ -137,53 +135,65 @@ private static Route countRecords(
...
+            return onComplete(Ask.getGoldenRecordAuditTrail(actorSystem, backEnd, gid),
+                    result -> {
+                        if (!result.isSuccess()) {
+                            final var e = result.failed().get();
+                            LOGGER.error(e.getLocalizedMessage(), e);
+                            return mapError(new MpiServiceError.InternalError(e.getLocalizedMessage()));
+                        }
+                        return complete(StatusCodes.OK, result.get().auditTrail(), JSON_MARSHALLER);
+                    });
+        } catch (NumberFormatException e) {
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java:174]: same as above

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java [hunk 8]
```diff
@@ -260,136 +270,147 @@ private static Route postCrFindSourceId(
...
-                                                 expandedGoldenRecords -> complete(StatusCodes.OK,
-                                                                                   expandedGoldenRecords.stream()
-                                                                                                        .map(ApiModels.ApiExpandedGoldenRecord::fromExpandedGoldenRecord)
-                                                                                                        .toList(),
-                                                                                   JSON_MARSHALLER));
-                           });
+      return entity(Jackson.unmarshaller(ApiModels.ApiNotifications.class), requestData -> {
+         try {
+            return onComplete(
+                  Ask.getNotifications(actorSystem, backEnd, requestData.limit(), requestData.offset(),
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java:279]: @sushantpatil1214 pass the entire request object

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java [hunk 9]
```diff
@@ -260,136 +270,147 @@ private static Route postCrFindSourceId(
...
+                        requestData.startDate(), requestData.endDate(), requestData.states()),
+                  result -> {
+                     if (!result.isSuccess()) {
+                        final var e = result.failed().get();
+                        LOGGER.error(e.getLocalizedMessage(), e);
+                        return mapError(new MpiServiceError.InternalError(e.getLocalizedMessage()));
+                     }
+                     return complete(StatusCodes.OK, result.get(), JSON_MARSHALLER);
+                  });
+         } catch (NumberFormatException e) {
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java:289]: same as previous mention

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java [hunk 10]
```diff
@@ -260,136 +270,147 @@ private static Route postCrFindSourceId(
...
-                                                                                   expandedGoldenRecords.stream()
-                                                                                                        .map(ApiModels.ApiExpandedGoldenRecord::fromExpandedGoldenRecord)
-                                                                                                        .toList(),
-                                                                                   JSON_MARSHALLER));
-                           });
-      });
-   }
 
-   private static Route getExpandedInteractionsUsingCSV(
+   private static Route postExpandedGoldenRecordsUsingParameterList(
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java:297]:
```suggestion
   private static Route getExpandedGoldenRecordsForUidList(
```

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java [hunk 11]
```diff
@@ -260,136 +270,147 @@ private static Route postCrFindSourceId(
...
-                                                           final var e = result.failed().get();
-                                                           LOGGER.error(e.getLocalizedMessage(), e);
-                                                           return mapError(new MpiServiceError.InternalError(e.getLocalizedMessage()));
-                                                        }
-                                                        return complete(StatusCodes.OK, result.get(), JSON_MARSHALLER);
-                                                     }))))));
-   }
-
-   private static Route getExpandedGoldenRecordsUsingParameterList(
+   private static Route postNotifications(
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java:273]:
```suggestion
   private static Route getNotifications(
```

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java [hunk 12]
```diff
@@ -225,7 +235,7 @@ private static Route countInteractions(
                         });
    }
 
-   private static Route getGidsAll(
+   private static Route postGidsAll(
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java:238]:
```suggestion
   private static Route getGidsAll(
```

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java [hunk 13]
```diff
@@ -137,53 +135,65 @@ private static Route countRecords(
...
-                                         result -> {
-                                            if (!result.isSuccess()) {
-                                               final var e = result.failed().get();
-                                               LOGGER.error(e.getLocalizedMessage(), e);
-                                               return mapError(new MpiServiceError.InternalError(e.getLocalizedMessage()));
-                                            }
-                                            return complete(StatusCodes.OK, result.get().auditTrail(), JSON_MARSHALLER);
-                                         }));
-   }
+   private static Route postGoldenRecordAuditTrail(
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java:159]:
```suggestion
   private static Route getGoldenRecordAuditTrail(
```

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java [hunk 14]
```diff
@@ -137,53 +135,65 @@ private static Route countRecords(
...
-                                                                result -> {
-                                                                   if (!result.isSuccess()) {
-                                                                      final var e = result.failed().get();
-                                                                      LOGGER.error(e.getLocalizedMessage(), e);
-                                                                      return mapError(new MpiServiceError.InternalError(e.getLocalizedMessage()));
-                                                                   }
-                                                                   return complete(StatusCodes.OK, result.get(), JSON_MARSHALLER);
-                                                                })));
-   }
+   private static Route postGidsPaged(
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/Routes.java:138]:
```suggestion
   private static Route getGidsPaged(
```

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/ProxyRoutes.java [hunk 15]
```diff
@@ -458,43 +458,59 @@ static Route proxyPatchCrUpdateFields(
...
-      final var uri = String.format(Locale.ROOT,
-                                    "http://%s:%d/JeMPI/%s?iid=%s",
-                                    linkerIP,
-                                    linkerPort,
-                                    GlobalConstants.SEGMENT_PROXY_GET_CANDIDATES_WITH_SCORES,
-                                    iid);
-      final var request = HttpRequest.create(uri);
-      final var stage = http.singleRequest(request);
-      return stage.thenApply(response -> response);
+   private static CompletionStage<HttpResponse> proxyPostCandidatesWithScoreDoIt(
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/ProxyRoutes.java:461]:
```suggestion
   private static CompletionStage<HttpResponse> proxyGetCandidatesWithScore(
```

File: JeMPI_Apps/JeMPI_Linker/src/main/java/org/jembi/jempi/linker/HttpServer.java [hunk 16]
```diff
@@ -63,7 +63,7 @@ private Route createRoute(
                                                        path(GlobalConstants.SEGMENT_PROXY_POST_CR_LINK_BY_SOURCE_ID_UPDATE,
                                                             () -> Routes.proxyPostCrLinkBySourceIdUpdate(actorSystem, backEnd)))),
                                      get(() -> concat(// path("mu", () -> Routes.routeMU(actorSystem, backEnd)),
-                                                      path(GlobalConstants.SEGMENT_PROXY_GET_CANDIDATES_WITH_SCORES,
+                                                      path(GlobalConstants.SEGMENT_PROXY_POST_CANDIDATES_WITH_SCORES,
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_Linker/src/main/java/org/jembi/jempi/linker/HttpServer.java:66]: This should move to post definitions

File: JeMPI_Apps/JeMPI_Linker/src/main/java/org/jembi/jempi/linker/Routes.java [hunk 17]
```diff
@@ -32,26 +32,33 @@ static StatusCode logHttpError(
...
+                        }
+                        return response.get()
+                                .candidates()
+                                .mapLeft(MapError::mapError)
+                                .fold(error -> error,
+                                        candidateList -> complete(StatusCodes.OK,
+                                                candidateList,
+                                                Jackson.marshaller()));
+                    });
+        } catch (NumberFormatException e) {
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_Linker/src/main/java/org/jembi/jempi/linker/Routes.java:55]: same as above reference

File: JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/GlobalConstants.java [hunk 18]
```diff
@@ -22,21 +22,21 @@ public final class GlobalConstants {
...
-   public static final String SEGMENT_PATCH_GOLDEN_RECORD = "goldenRecord";
-   public static final String SEGMENT_POST_IID_GID_LINK = "link";
-   public static final String SEGMENT_POST_IID_NEW_GID_LINK = "unLink";
+   public static final String SEGMENT_POST_GIDS_ALL = "gidsAll";
+   public static final String SEGMENT_POST_GIDS_PAGED = "gidsPaged";
+   public static final String SEGMENT_POST_INTERACTION = "interaction";
+   public static final String SEGMENT_POST_EXPANDED_GOLDEN_RECORD = "expandedGoldenRecord";
+   public static final String SEGMENT_POST_EXPANDED_GOLDEN_RECORDS_USING_PARAMETER_LIST = "expandedGoldenRecords";
+   public static final String SEGMENT_POST_EXPANDED_GOLDEN_RECORDS_USING_CSV = "expandedGoldenRecordsCsv";
+   public static final String SEGMENT_POST_EXPANDED_INTERACTIONS_USING_CSV = "expandedInteractionsCsv";
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/GlobalConstants.java:31]:
```suggestion
   public static final String SEGMENT_POST_EXPANDED_GOLDEN_RECORDS_FOR_GOLDEN_IDS= "expandedGoldenRecordsForGoldenIds";
   public static final String SEGMENT_POST_EXPANDED_INTERACTIONS_FOR_INTERACTION_IDS = "expandedInteractionsForInteractionIds";
```

File: JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/GlobalConstants.java [hunk 19]
```diff
@@ -22,21 +22,21 @@ public final class GlobalConstants {
...
+   public static final String SEGMENT_POST_GOLDEN_RECORD_AUDIT_TRAIL = "goldenRecordAuditTrail";
+   public static final String SEGMENT_POST_INTERACTION_AUDIT_TRAIL = "interactionAuditTrail";
+   public static final String SEGMENT_POST_FIELDS_CONFIG = "config";
+   public static final String SEGMENT_POST_LINKED_RECORDS = "linkedRecords";
+   public static final String SEGMENT_POST_NOTIFICATIONS = "matchesForReview";
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/GlobalConstants.java:36]:
```suggestion
   public static final String SEGMENT_POST_NOTIFICATIONS = "notifications";
```

File: JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/GlobalConstants.java [hunk 20]
```diff
@@ -22,21 +22,21 @@ public final class GlobalConstants {
...
+   public static final String SEGMENT_POST_GOLDEN_RECORD = "updateGoldenRecordFieldsForId";
+   public static final String SEGMENT_POST_IID_GID_LINK = "relink";
+   public static final String SEGMENT_POST_IID_NEW_GID_LINK = "newLink";
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/GlobalConstants.java:39]:
```suggestion
   public static final String SEGMENT_POST_RELINK = "relink";
   public static final String SEGMENT_POST_NEW_LINK = "newLink";
```

File: JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/GlobalConstants.java [hunk 21]
```diff
@@ -49,13 +49,13 @@ public final class GlobalConstants {
    public static final String SEGMENT_PROXY_POST_CR_LINK_BY_SOURCE_ID_UPDATE = "crLinkBySourceIdUpdate";
    public static final String SEGMENT_PROXY_POST_CR_FIND = "crFind";
    public static final String SEGMENT_PROXY_POST_CR_CANDIDATES = "crCandidates";
-   public static final String SEGMENT_PROXY_PATCH_CR_UPDATE_FIELDS = "crUpdateFields";
+   public static final String SEGMENT_PROXY_POST_CR_UPDATE_FIELDS = "crUpdateFields";
    public static final String SEGMENT_POST_CR_FIND_SOURCE_ID = "crFindSourceId";
-   public static final String SEGMENT_PROXY_GET_CANDIDATES_WITH_SCORES = "candidateGoldenRecords";
-   public static final String SEGMENT_PROXY_POST_CALCULATE_SCORES = "calculateScores";
-   public static final String SEGMENT_PROXY_GET_DASHBOARD_DATA = "dashboardData";
+   public static final String SEGMENT_PROXY_POST_CANDIDATES_WITH_SCORES = "candidateGoldenRecords";
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/GlobalConstants.java:54]:
```suggestion
   public static final String SEGMENT_PROXY_POST_CANDIDATE_GOLDEN_RECORDS

Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/GlobalConstants.java:54]:
```suggestion
   public static final String SEGMENT_PROXY_POST_CANDIDATE_GOLDEN_RECORDS = "candidateGoldenRecords";
```

File: JeMPI_Apps/JeMPI_Linker/src/main/java/org/jembi/jempi/linker/HttpServer.java [hunk 22]
```diff
@@ -42,13 +42,13 @@ private Route createRoute(
          final ActorSystem<Void> actorSystem,
          final ActorRef<BackEnd.Request> backEnd) {
       return pathPrefix("JeMPI",
-                        () -> concat(patch(() -> path(GlobalConstants.SEGMENT_PROXY_PATCH_CR_UPDATE_FIELDS,
+                        () -> concat(patch(() -> path(GlobalConstants.SEGMENT_PROXY_POST_CR_UPDATE_FIELDS,
```
[the remark this answers is not in the document]
Comment (sushantpatil1214) [reply JeMPI_Apps/JeMPI_Linker/src/main/java/org/jembi/jempi/linker/HttpServer.java:45]: This is commented section not used anywhere

**Pull request merged**